src/Microsoft.IdentityModel.Tokens/EventBasedLRUCache.cs

```diff
@@ -222,10 +238,14 @@ public bool SetValue(TKey key, TValue value, DateTime expirationTime)
                 var newCacheItem = new LRUCacheItem<TKey, TValue>(key, value, expirationTime);
                 _eventQueue.Add(() =>
                 {
+                    // if maintaining a LRU move item to front
                     // Add a remove operation in case two threads are trying to add the same value. Only the second remove will succeed in this case.
-                    _doubleLinkedList.Remove(newCacheItem);
+                    if (_maintainLRU)
+                        _doubleLinkedList.Remove(newCacheItem);
```

Comment:
@brentschmaltz As Keegan mentioned, we'd need to always remove in this case. This whole 'else' block will only be executed when the value **doesn't** already exist in the cache, so it isn't actually being moved to the front here.

The remove is here in case two threads adding the same value happen to enter this block of code at the same time. We need a remove in order to get rid of the duplicate in this sort of situation.